### PR TITLE
Reverted practice of automatically setting cloudProvider.external for some providers

### LINKED ISSUE #2190: Some cloud providers are defaulted to external
### What happened?

While working on generated e2e tests, some providers has been defaulted to be external regardless of actual flag.

### Expected behavior

Old behavior should be preserved for API compatibility.

TODO:

* [ ] Drop defaulting to external
* [ ] Add additional generated e2e tests for `external: true` case

### PR DESCRIPTION
**Pull request by kron4eg**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Fixes an unreleased defaulting mistake + some new generated e2e tests

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2190

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Reverted practice of automatically setting cloudProvider.external for some providers
```

### CODE REVIEW

File: testv2/tests.yml [hunk 1]
```diff
@@ -320,3 +275,159 @@
...
+    - name: digitalocean_default
+    - name: equinixmetal_default
+    - name: hetzner_default
+    - name: openstack_default
+    - name: vsphere_default
+    - name: vsphere_flatcar
+
+- scenario: upgrade_containerd_external
+  initVersion: v1.22.11
+  upgradedVersion: v1.23.8
```
Comment (xmudrii) [inline testv2/tests.yml:401]: Why don't we have 1.23 to 1.24 upgrade tests?

**Review by xmudrii — COMMENTED** [review]
Just a nit, otherwise LGTM.

File: testv2/tests.yml (hunk 1 above)
Comment (kron4eg) [reply testv2/tests.yml:401]: it was missing for non external containerd too, probably it was too fresh version at a time and later forgotten to be added :shrug: 

will add (+ external conformance too)

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**